idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+
+;====finds how many pixels old and new chs share====== 
+				clone[j]=n_elements(setintersection(where(prevrot eq i),where(iarr eq j)))
+
+			endif
+		endfor
+
+;=====defines which new ch is most likely a previously segmented ch======
+		loc=where(clone eq max(clone))
+		if loc[0] ne 0 and clone[loc[0]] gt 1 then begin
+			segarr[where(iarr eq loc[0])]=i
+
+;=====this ch cannot be reclassified========
+			taken[loc[0]]=1
+
+;====contour and label ch with tracked number========
+			Contour,segarr,ax,ay,/over,levels=[max(segarr)-0.5],path_xy=xy,path_info=info,/path_data_coords;,color='FFFFFF'xL,thick=3
+			for j=0L,(n_elements(info)-1) do begin
+				chpts = [FINDGEN(info(J).N), 0]
+				PLOTS, xy(*,INFO(J).OFFSET + (10*chpts[0:(max(chpts)/10)]) ), /DATA,color='FFFFFF'xL,thick=3
+				PLOTS, xy(*,INFO(J).OFFSET), /DATA,color='FFFFFF'xL,thick=2,/continue
+			endfor	
+			cenarr[*]=0
+			cenarr[where(segarr eq max(segarr))]=1
+			precen=centroid(cenarr)
+
+			ch=[ch,long(max(segarr))]
+			n=[n,n_elements(where(segarr eq max(segarr)))]
+			x=[x,xco[where(segarr eq max(segarr))]]
+			y=[y,yco[where(segarr eq max(segarr))]]
+
+			XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='000000'xL,charthick=7,charsize=2
+			XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='FFFFFF'xL,charthick=3,charsize=2
+			props[0,loc[0]+1]=strcompress(long(max(segarr)),/remove_all)
+
+		endif
+	endfor
+endif
+
+;====cycle through any CHs which was not relabeled in tracking========
+for i=1,ident do begin
+	if taken[i] ne 1 then begin
+		mxseg++
+		segarr[where(iarr eq i)]=mxseg
+
+;====contour and label CH boundary======
+		Contour,segarr,ax,ay,/over,levels=[mxseg-0.5],path_xy=xy,path_info=info,/path_data_coords;,color='FFFFFF'xL,thick=3
+		for j=0L,(n_elements(info)-1) do begin
+			chpts = [FINDGEN(info(J).N), 0]
+			PLOTS, xy(*,INFO(J).OFFSET + (10*chpts[0:(max(chpts)/10)]) ), /DATA,color='FFFFFF'xL,thick=3
+			PLOTS, xy(*,INFO(J).OFFSET), /DATA,color='FFFFFF'xL,thick=2,/continue		
+		endfor	
+		cenarr[*]=0
+		cenarr[where(segarr eq max(segarr))]=1
+		precen=centroid(cenarr)
+
+		ch=[ch,long(max(segarr))]
+		n=[n,n_elements(where(segarr eq max(segarr)))]
+		x=[x,xco[where(segarr eq max(segarr))]]
+		y=[y,yco[where(segarr eq max(segarr))]]
+
+		XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='000000'xL,charthick=7,charsize=2
+		XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='FFFFFF'xL,charthick=3,charsize=2
+		props[0,i+1]=strcompress(long(max(segarr)),/remove_all)
+	endif
+endfor
+
+;====display off-limb CH regions=======
+Contour,offarr,ax,ay,/over,levels=[0.5],color='FFFFFF'xL
+
+;====create image in output folder=======
+void = cgSnapshot(Position=[0.035,0.035,0.98,0.98],File=outpath+'/pngs/saia/saia_chimr_ch_'+time2file(map[1].time,/seconds)+'_pre', /PNG, /NoDialog)
```

Comment:
I would avoid to save directly from the screen. SolarMonitor pipeline should be not using any X device, only plotting on the Z buffer and saving from there.